UnitTests/UnitTests.XamlIslands/XamlIslandsTest_ThemeListener_Threading.cs

```diff
@@ -0,0 +1,53 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Microsoft.Toolkit.Uwp.UI.Helpers;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Windows.UI.Xaml;
+
+namespace UnitTests.XamlIslands
+{
+    [STATestClass]
+    public partial class XamlIslandsTest_ThemeListener_Threading
```

Comment:
>XamlIslandsTest_ThemeListener_Threading [](start = 25, length = 39)

you may want to spin up two different UI threads (each on their own STA) to check that you dispatch back to the correct UI thread in both cases, as a test.